core/services/adapters/PdfAdapter.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace EventEspresso\core\services\adapters;
+
+use Dompdf\Autoloader;
+use Dompdf\Dompdf;
+use Dompdf\Options;
+
+/**
+ * Class PdfAdapter
+ *
+ * @package EventEspresso\core\services\adapters
+ */
+class PdfAdapter
+{
+    /**
+     * @var Options|null
+     */
+    private $options;
+
+    /**
+     * @var string
+     */
+    private $content;
+
+
+    /**
+     * PdfAdapter constructor.
+     */
+    public function __construct()
+    {
+        $this->options = null;
+        $this->content = '';
+    }
+
+
+    /**
+     * @return Options|null
+     */
+    public function getOptions()
+    {
+        // only load dompdf if nobody else has yet...
+        if (! class_exists('Dompdf\Autoloader') and is_readable(EE_THIRD_PARTY . 'dompdf/src/Autoloader.php')) {
+            require_once(EE_THIRD_PARTY . 'dompdf/src/Autoloader.php');
+            Autoloader::register();
+        }
+        if (! class_exists('Dompdf\Options')) {
+            return null;
+        }
+        $options = new Options();
+        $options->set('isRemoteEnabled', true);
+        $options->set('isJavascriptEnabled', false);
+        if (defined('DOMPDF_FONT_DIR')) {
+            $options->setFontDir(DOMPDF_FONT_DIR);
+            $options->setFontCache(DOMPDF_FONT_DIR);
+        }
+        // Allow changing the paper size.
+        if (defined('DOMPDF_DEFAULT_PAPER_SIZE')) {
+            $options->set('defaultPaperSize', DOMPDF_DEFAULT_PAPER_SIZE);
+        }
+        return $options;
+    }
+
+
+    /**
+     * @param Options $options
+     * @return $this
+     */
+    public function setOptions(Options $options): self
```

Comment:
can you plz verify that `self` is an acceptable return value in PHP 7.1?
I prefer we use the actual classname (in phpdocs too), because sometimes you may be looking at a code snippet posted in chat or elsewhere, and it's impossible to know what class `self` represents since you don't have your IDE to help you out, whereas seeing:

```php
     * @return PdfAdapter
     */
    public function setOptions(Options $options): PdfAdapter
```

is immediately understandable